vyper/parser/expr.py

```diff
@@ -330,9 +330,10 @@ def arithmetic(self):
             new_unit = combine_units(left.typ.unit, right.typ.unit)
             if ltyp == rtyp == 'uint256':
                 o = LLLnode.from_list(['seq',
-                                        # Checks that: a == 0 || a / b == b
-                                        ['assert', ['or', ['iszero', left],
-                                        ['eq', ['div', ['mul', left, right], left], right]]],
+                                        # Checks that: a == 0
+                                        # Gas optimization: this is cheaper than asserting 'a' not being zero, but the
+                                        # benefit is lost if 'b' is also tested.
+                                        ['assert', ['iszero', left]],
                                         ['mul', left, right]], typ=BaseType('uint256', new_unit), pos=getpos(self.expr))
```

Comment:
Looking at the solidity contract:
```
    if (a == 0) {
      return 0;
    }

    c = a * b;
    assert(c / a == b);
    return c
```
We are only left with `'assert', ['iszero', left]],` .
But the second assert with c / a * b still needs to occur?

Reply:
That's also why the CI keeps failing.

Reply:
If you want to emulate the the return 0 it would be, something like: 
```
['if', 
   ['eq', left, 0], 
   [0],
   ['seq'
       ['assert', ['eq', ['div', ['mul', left, right], left], right] ]],
       ['mul', left, right]]
]
```

Reply:
Thank you @jacqueswww I will test it and implement it right away.